src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -834,6 +869,18 @@ private bool InferConnectionDetails()
                     ReportSNIError(SNIProviders.INVALID_PROV);
                     return false;
                 }
+
+                if (ConnectionProtocol == DataSource.Protocol.None)
```

Comment:
Currently TCP explicitly assigned by DataSource when there is no protocol in connection string. And it creates TCP handle in `CreateConnectionHandle()` when there is no protocol. This is TCP default policy scattered over the codes, which should be kept and managed in one place.